### PR TITLE
WT-4033 Add ability to alter tables without taking exclusive access

### PR DESCRIPTION
**Pull request by agorrod**
This particular commit changes which object types app_metadata appears in - we want it to be in only table, so we can alter without exclusive access.

### CODE REVIEW

File: dist/api_data.py [hunk 1]
```diff
@@ -357,7 +352,17 @@ def __cmp__(self, other):
...
+# objects. They can be safely altered without exclusive access, as long as
+# the user doesn't need to see refreshed values without a restart.
+table_runtime_config = [
+    Config('app_metadata', '', r'''
+        application-owned metadata for this object'''),
+]
+
+# The table runtime metadata also applied to LSM trees, it currently contains
+# the user configurable app_metadata.
+lsm_meta = file_config + lsm_config + table_runtime_config + [
```
Comment (michaelcahill) [inline dist/api_data.py]: Why do LSM trees need {{app_metadata}} if btrees don't?

Comment (sulabhM) [reply dist/api_data.py]: We thought table: and lsm: are the two main schemas users (for instance Mongodb) operate and set app_metadata on. So we decided to include LSM too.

**Comment by sulabhM** [comment]
autobuild: retest this please

File: dist/api_data.py [hunk 2]
```diff
@@ -938,7 +938,15 @@ def __cmp__(self, other):
 
 'WT_CURSOR.reconfigure' : Method(cursor_runtime_config),
 
-'WT_SESSION.alter' : Method(file_runtime_config),
+'WT_SESSION.alter' : Method(file_runtime_config + [
+    Config('exclusive_refreshed', 'true', r'''
```
Comment (sulabhM) [inline dist/api_data.py]: Not too happy with the name, suggestions welcome.

**Comment by sulabhM** [comment]
@agorrod , this is ready for review.

**Comment by agorrod** [comment]
Thanks @sulabhM this change lgtm - I've made some wording updates. If it passes testing I'd be happy for it to be merged.

**Comment by sulabhM** [comment]
autobuild: retest this please

**Pull request merged**